### DISCUSSION TITLE
Install <winrt/Microsoft.UI.Composition.SystemBackdrops.h>

### DISCUSSION THREAD

**Discussion by r2d2Proton** [microsoft/WindowsAppSDK#2616] [General]
How do I download/install the Microsoft.UI components?

Thank you for the help.

**Comment by castorix** [comment]
They are generated when you build a CPP WinUI project (in .\Generated Files\winrt)

**Reply by r2d2Proton** [reply]
Thank you for the reply.
I downloaded https://www.nuget.org/packages/Microsoft.WindowsAppSDK
Renamed it to a .zip and extracted it.

Then I ran a few commands:
cppwinrt -input sdk -output "<dirName>"
cppwinrt -input "lib\uap10.0.18362\Microsoft.UI.winmd" -input sdk -output "<dirName>"
:
cppwinrt -input "lib\uap10.0\Microsoft.Windows.System.winmd" -input sdk -output "<dirName>"

I am looking for windows.ui.composition.interop.h.

By chance, do you know where I can download it?

Thank you again.

**Reply by castorix** [reply]
> I am looking for windows.ui.composition.interop.h.
> By chance, do you know where I can download it?

It is included in the [Windows SDK](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/)
(on my version (Windows 10), once installed, I have it in _C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0_\ )

**Reply by r2d2Proton** [reply]
Thank you for the link. I just completed a VS2022 Repair, which apparently reset a few things. I am not sure what VS installs. But at least I can download and reinstall the Windows SDK. I forgot about that link. Thanks.
